js/filter.js

```diff
@@ -107,17 +111,22 @@ export const filterToParams = (
             ...flatMap(
                 ([field, operator]) =>
                     _buildFilter(field, operator, filterS, keywordFields, { keywords: keywords }),
-                Object.entries(imperfectFilterFields)
+                Object.entries(imperfectFilterFields).filter(
+                    ([field, _]) => !removeFunc[field] || !removeFunc[field](filterS)

```

Comment:
not sure if we should address it in this PR,  but could we rename this `filterS` to something else as to differentiate it from `filters`